aioshelly/block_device/device.py

```diff
@@ -421,6 +421,21 @@ async def trigger_reboot(self) -> None:
         """Trigger a device reboot."""
         await self.http_request("get", "reboot")
 
+    async def set_auth(
+        self,
+        enabled: bool,
+        username: str | None = None,
+        password: str | None = None,
+    ) -> dict[str, Any]:
+        """Set authentication on the device."""
+        if enabled:
+            if username is None or password is None:
+                raise ValueError("username and password required when enabling auth")
+            params = {"enabled": 1, "username": username, "password": password}
+        else:
+            params = {"enabled": 0}
```

Comment:
According to [Shelly API](https://github.com/home-assistant/core/issues/155884#issuecomment-3498882445) this should be a Boolean with `True` or `False`